Ingress with no http rull, occurs nil pointer exception in cdi-deployment

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
If we create ingress with no http rule, and deploy CDI, occurs nil pointer exception. Because of http rule is optional field, no http rule ingress is allowed.

ingress is following
```yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: test-ingress
spec:
  rules:
  - http:
```

When deploying CDI, the cdi-deployment deploy failed ans log is following.

```shell
I1206 05:33:36.036032       1 leaderelection.go:214] successfully acquired lease cdi/cdi-controller-leader-election-helper
I1206 05:33:36.036168       1 leaderelection.go:103] Successfully acquired leadership lease
I1206 05:33:36.036178       1 controller.go:98] Starting CDI controller components
I1206 05:33:36.037934       1 controller.go:205] created cdi controllers
I1206 05:33:36.042641       1 keystore.go:79] Retrieved CA key/cert server.upload.cdi.kubevirt.io from kubernetes
I1206 05:33:36.044914       1 keystore.go:79] Retrieved CA key/cert client.upload-server.cdi.kubevirt.io from kubernetes
I1206 05:33:36.046977       1 keystore.go:161] Retrieved client key/cert client.upload-server.cdi.kubevirt.io from kubernetes
I1206 05:33:36.048781       1 keystore.go:79] Retrieved CA key/cert proxy.upload.cdi.kubevirt.io from kubernetes
I1206 05:33:36.050564       1 keystore.go:123] Retrieved server key/cert cdi-uploadproxy.cdi from kubernetes
E1206 05:33:36.059521       1 config-controller.go:148] error getting CDI config: cdiconfig.cdi.kubevirt.io "config" not found
E1206 05:33:36.060016       1 config-controller.go:148] error getting CDI config: cdiconfig.cdi.kubevirt.io "config" not found
I1206 05:33:36.060169       1 controller.go:229] started informers
I1206 05:33:36.163232       1 upload-controller.go:206] Started workers
I1206 05:33:36.168418       1 config-controller.go:361] Started workers
E1206 05:33:36.168582       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/gimme/.gimme/versions/go1.11.5.linux.amd64/src/runtime/asm_amd64.s:522
/gimme/.gimme/versions/go1.11.5.linux.amd64/src/runtime/panic.go:513
/gimme/.gimme/versions/go1.11.5.linux.amd64/src/runtime/panic.go:82
/gimme/.gimme/versions/go1.11.5.linux.amd64/src/runtime/signal_unix.go:390
/go/src/kubevirt.io/containerized-data-importer/pkg/controller/util.go:1290
/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:255
/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:202
/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:210
/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:179
/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:358
/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133
/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134
/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/gimme/.gimme/versions/go1.11.5.linux.amd64/src/runtime/asm_amd64.s:1333
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x10b1101]

goroutine 111 [running]:
kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
panic(0x11f5c60, 0x2070300)
	/gimme/.gimme/versions/go1.11.5.linux.amd64/src/runtime/panic.go:513 +0x1b9
kubevirt.io/containerized-data-importer/pkg/controller.getURLFromIngress(0xc0000fa2c0, 0xc000040044, 0xf, 0xc0007e1040, 0x2)
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/util.go:1290 +0x111
kubevirt.io/containerized-data-importer/pkg/controller.(*ConfigController).syncHandler(0xc0004c20f0, 0xc0004510f8, 0x6, 0x118b360, 0xc0007e10e0)
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:255 +0x65e
kubevirt.io/containerized-data-importer/pkg/controller.(*ConfigController).processNextWorkItem.func1(0xc0004c20f0, 0x118b360, 0xc0007e10e0, 0x0, 0x0)
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:202 +0xda
kubevirt.io/containerized-data-importer/pkg/controller.(*ConfigController).processNextWorkItem(0xc0004c20f0, 0xc00065a350)
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:210 +0x50
kubevirt.io/containerized-data-importer/pkg/controller.(*ConfigController).runWorker(0xc0004c20f0)
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:179 +0x2b
kubevirt.io/containerized-data-importer/pkg/controller.(*ConfigController).runWorker-fm()
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:358 +0x2a
kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0007e1010)
	/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0007e1010, 0x3b9aca00, 0x0, 0x1, 0xc00009e900)
	/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbe
kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0007e1010, 0x3b9aca00, 0xc00009e900)
	/go/src/kubevirt.io/containerized-data-importer/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by kubevirt.io/containerized-data-importer/pkg/controller.(*ConfigController).Run
	/go/src/kubevirt.io/containerized-data-importer/pkg/controller/config-controller.go:358 +0x1e6
```

**What you expected to happen**:
CDI running

**How to reproduce it (as minimally and precisely as possible)**:
- creating ingress with no http
- deploying CDI

**Anything else we need to know?**:
NONE

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`): v1.11.0
- Kubernetes version (use `kubectl version`): v1.16.2
- Cloud provider or hardware configuration: Minikube
- Install tools: Minikube
- Others:

--- a/go.mod
+++ b/go.mod
@@ -14,6 +14,7 @@ require (
 	github.com/go-logr/logr v0.0.0-20190813230443-d63354a31b29
 	github.com/go-logr/zapr v0.0.0-20190813212058-2e515ec1daf7 // indirect
 	github.com/go-openapi/spec v0.19.2
+	github.com/go-openapi/validate v0.18.0 // indirect
 	github.com/gogo/protobuf v1.2.1 // indirect
 	github.com/golang/groupcache v0.0.0-20190129154638-5b532d6fd5ef // indirect
 	github.com/golang/protobuf v1.3.1 // indirect
@@ -43,6 +44,7 @@ require (
 	golang.org/x/net v0.0.0-20191007182048-72f939374954 // indirect
 	golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a // indirect
 	golang.org/x/sys v0.0.0-20191008105621-543471e840be
+	google.golang.org/genproto v0.0.0-20181016170114-94acd270e44e // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
 	gopkg.in/ini.v1 v1.48.0 // indirect
 	gopkg.in/square/go-jose.v2 v2.3.1

--- a/pkg/controller/config-controller.go
+++ b/pkg/controller/config-controller.go
@@ -151,6 +151,10 @@ func (c *ConfigController) handleObject(obj interface{}) {
 
 	if ing, ok := obj.(*extensionsv1beta1.Ingress); ok {
 		for _, rule := range ing.Spec.Rules {
+			if rule.HTTP == nil {
+				continue
+			}
+
 			for _, path := range rule.HTTP.Paths {
 				if path.Backend.ServiceName == c.uploadProxyServiceName {
 					c.enqueueCDIConfig(config)

